pkgs/development/libraries/libjxl/default.nix

```diff
@@ -0,0 +1,106 @@
+{ stdenv, lib, fetchFromGitHub
+, asciidoc
+, brotli
+, cmake
+, graphviz
+, doxygen
+, giflib
+, gperftools
+, gtest
+, libjpeg
+, libpng
+, libwebp
+, openexr
+, pkgconfig
+, python3
+, zlib
+}:
+
+stdenv.mkDerivation rec {
+
+  pname = "libjxl";
+  version = "2021-06-22-409efe02";
+
+  src = fetchFromGitHub {
+    owner = "libjxl";
+    repo = "libjxl";
+    rev = "409efe027d6a4a4446b84abe8d7b2fa40409257d";
+    sha256 = "1akb6yyp2h4h6mfcqd4bgr3ybcik5v5kdc1rxaqnyjs7fp2f6nvv";
+    # There are various submodules in `third_party/`.
+    fetchSubmodules = true;
```

Comment:
Those should be probably devondered.

Reply:
See previous comment from https://github.com/NixOS/nixpkgs/pull/103160#discussion_r519525156

I have de-vendored `brotli`, but `libjxl` has many other submodules (https://github.com/libjxl/libjxl/tree/409efe027d6a4a4446b84abe8d7b2fa40409257d/third_party) that are frequently updated to pre-release versions, so I would want to put effort into that not before `libjxl` has become more stable.

I've added a code comment about that.